src/v/cloud_storage/remote.cc

```diff
@@ -53,13 +53,45 @@ remote::remote(
         std::visit([](auto&& cfg) { return cfg.disable_metrics; }, conf))),
       remote_metrics_disabled(static_cast<bool>(std::visit(
         [](auto&& cfg) { return cfg.disable_public_metrics; }, conf))),
-      *_materialized) {
-    // If the credentials source is from config file, bypass the background op
-    // to refresh credentials periodically, and load pool with static
+      *_materialized)
+  , _azure_shared_key_binding(
+      config::shard_local_cfg().cloud_storage_azure_shared_key.bind()) {
+    // If the credentials source is from config file, bypass the background
+    // op to refresh credentials periodically, and load pool with static
     // credentials right now.
     if (_auth_refresh_bg_op.is_static_config()) {
         _pool.load_credentials(_auth_refresh_bg_op.build_static_credentials());
     }
+
+    _azure_shared_key_binding.watch([this] {
+        auto current_config = _auth_refresh_bg_op.get_client_config();
+        if (!std::holds_alternative<cloud_storage_clients::abs_configuration>(
+              current_config)) {
+            return;
```

Comment:
Nit: maybe we should log here? This seems like an anomaly if the code ends up here. Perhaps as a result of user error (setting the abs key in s3 env)? 

Although in that case logging wouldn't help much anyway so maybe this is fine as is.

Reply:
Setting the abs key while using an S3 configuration should be harmless, although very strange. I think it's a good idea to log here.